rackcity/models/asset_change_planner.py

```diff
@@ -0,0 +1,79 @@
+from django.db import models
+from .asset import validate_hostname, validate_owner
+from django.core.validators import MaxValueValidator, MinValueValidator
+from django.core.exceptions import ValidationError, ObjectDoesNotExist
+from .it_model import ITModel
+from .rack import Rack
+from .asset import Asset
+from .change_plan import ChangePlan
+
+
+class AssetCP(models.Model):
+    related_asset = models.ForeignKey(
+        Asset,
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+    )
+    ## TODO: related_decomissioned_asset
+    is_conflict = models.BooleanField(
+        default=False,
+        blank=True,
+    )
+    change_plan = models.ForeignKey(
+        ChangePlan,
+        on_delete=models.CASCADE,
+    )
+    asset_number = models.IntegerField(
+        unique=True,
+        validators=[
+            MinValueValidator(100000),
+            MaxValueValidator(999999)
+        ],
+        blank=True,
+        # If blank, Asset numbers will be assigned on execution
+        null=True,
+    )
+    hostname = models.CharField(
+        max_length=150,
+        unique=True,
+        validators=[validate_hostname],
+        null=True,
+        blank=True,
+    )
+    rack_position = models.PositiveIntegerField()
+    model = models.ForeignKey(
+        ITModel,
+        on_delete=models.CASCADE,
+        verbose_name="related model",
+    )
```

Comment:
FYSA, `on_delete=models.CASCADE` means that if a model is deleted, any AssetCP with that model will also be deleted without warning. I'm not sure what the desired behavior is there but if it's left as `CASCADE` then you'll likely want to add something to the check we do for existing assets when deleting a model

Reply:
changing this to SET_NULL, for both model and rack.